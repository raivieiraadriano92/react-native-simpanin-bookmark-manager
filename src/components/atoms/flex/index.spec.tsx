import React from 'react'
import { Text } from 'react-native'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import normalize from 'react-native-normalize'

import DefaultTheme from 'src/theme'

import Flex from './index'

describe('atoms / Flex', () => {
  it('should be able to render correctly', () => {
    const { findByText, getByTestId, toJSON } = render(
      <ThemeProvider theme={DefaultTheme}>
        <Flex
          flex
          margin="medium"
          marginBottom="medium"
          marginHorizontal="medium"
          marginLeft="medium"
          marginRight="medium"
          marginTop="medium"
          marginVertical="medium"
          padding="medium"
          paddingBottom="medium"
          paddingHorizontal="medium"
          paddingLeft="medium"
          paddingRight="medium"
          paddingTop="medium"
          paddingVertical="medium"
          style={{ backgroundColor: DefaultTheme.colors.primary }}
        >
          <Text>test</Text>
        </Flex>
      </ThemeProvider>
    )

    const json = toJSON()

    const component = getByTestId('container')

    expect(json).toMatchSnapshot()

    expect(findByText('test')).toBeTruthy()

    expect({
      ...component.props.style[0],
      ...component.props.style[1]
    }).toMatchObject({
      backgroundColor: DefaultTheme.colors.primary,
      flex: 1,
      margin: normalize(DefaultTheme.spacing.medium),
      marginBottom: normalize(DefaultTheme.spacing.medium, 'height'),
      marginHorizontal: normalize(DefaultTheme.spacing.medium),
      marginLeft: normalize(DefaultTheme.spacing.medium),
      marginRight: normalize(DefaultTheme.spacing.medium),
      marginTop: normalize(DefaultTheme.spacing.medium, 'height'),
      marginVertical: normalize(DefaultTheme.spacing.medium, 'height'),
      padding: normalize(DefaultTheme.spacing.medium),
      paddingBottom: normalize(DefaultTheme.spacing.medium, 'height'),
      paddingHorizontal: normalize(DefaultTheme.spacing.medium),
      paddingLeft: normalize(DefaultTheme.spacing.medium),
      paddingRight: normalize(DefaultTheme.spacing.medium),
      paddingTop: normalize(DefaultTheme.spacing.medium, 'height'),
      paddingVertical: normalize(DefaultTheme.spacing.medium, 'height')
    })
  })

  it('should be able to render without props', () => {
    const { findByText, toJSON } = render(
      <ThemeProvider theme={DefaultTheme}>
        <Flex />
      </ThemeProvider>
    )

    const json = toJSON()

    expect(json).toMatchSnapshot()

    expect(findByText('test')).toBeTruthy()
  })
})
