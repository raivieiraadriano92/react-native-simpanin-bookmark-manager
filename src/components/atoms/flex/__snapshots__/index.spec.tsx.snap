// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms / Flex should be able to render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "margin": 48,
        "marginBottom": 48,
        "marginHorizontal": 48,
        "marginLeft": 48,
        "marginRight": 48,
        "marginTop": 48,
        "marginVertical": 48,
        "padding": 48,
        "paddingBottom": 48,
        "paddingHorizontal": 48,
        "paddingLeft": 48,
        "paddingRight": 48,
        "paddingTop": 48,
        "paddingVertical": 48,
      },
      Object {
        "backgroundColor": "#5460FE",
      },
    ]
  }
  testID="container"
>
  <Text>
    test
  </Text>
</View>
`;

exports[`atoms / Flex should be able to render without props 1`] = `
<View
  style={Object {}}
  testID="container"
/>
`;
